app/models/framework/definition/ast/semantic_checker.rb

```diff
@@ -77,9 +77,19 @@ def raise_when_lookup_reference_invalid(field)
         end
 
         def raise_when_management_field_invalid(info)
-          referenced_field_name = [info.dig(:column_based, :column_names), info.dig(:flat_rate, :column)].compact.first(&:present?)
-          return if referenced_field_name.nil?
+          management_charge_modifier = [info.dig(:column_based, :column_names), info.dig(:flat_rate, :column)].compact.first(&:present?)
+          return if management_charge_modifier.blank?
 
+          if management_charge_modifier.is_a?(Array)
+            management_charge_modifier.each do |referenced_field_name|
```

Comment:
How about doing `Array(management_charge_modifier).each do |...|`, and then you could remove the conditional?